velox/exec/tests/utils/Cursor.cpp

```diff
@@ -230,10 +274,124 @@ bool TaskCursor::moveNext() {
   return current_ != nullptr;
 }
 
-bool TaskCursor::hasNext() {
+bool MultiThreadedTaskCursor::hasNext() {
   return queue_->hasNext();
 }
 
+class SingleThreadedTaskCursor : public TaskCursor {
+ public:
+  explicit SingleThreadedTaskCursor(const CursorParameters& params) {
+    VELOX_DCHECK(params.singleThreaded)
+    std::shared_ptr<core::QueryCtx> queryCtx;
+    if (params.queryCtx) {
+      queryCtx = params.queryCtx;
+    } else {
+      static std::atomic<uint64_t> cursorQueryId{0};
```

Comment:
It looks like there is a bit of copy-paste here. Would it be possible to refactor to avoid that?

Reply:
I've added a base class `TaskCursorBase` in the latest commit.

Actually I was worried multiple layers of class hierarchy might mess up the code, but it seems just fine after done.